mypy/plugin.py

```diff
@@ -332,3 +374,53 @@ def int_pow_callback(ctx: MethodContext) -> Type:
         else:
             return ctx.api.named_generic_type('builtins.float', [])
     return ctx.default_return_type
+
+
+def add_method(
+        info: TypeInfo,
+        method_name: str,
+        args: List[Argument],
+        ret_type: Type,
+        self_type: Type,
+        function_type: Instance) -> None:
+    from mypy.semanal import set_callable_name
+
+    first = [Argument(Var('self'), self_type, None, ARG_POS)]
+    args = first + args
+
+    arg_types = [arg.type_annotation for arg in args]
+    arg_names = [arg.variable.name() for arg in args]
+    arg_kinds = [arg.kind for arg in args]
+    assert None not in arg_types
+    signature = CallableType(arg_types, arg_kinds, arg_names,
+                             ret_type, function_type)
+    func = FuncDef(method_name, args, Block([]))
+    func.info = info
+    func.is_class = False
+    func.type = set_callable_name(signature, func)
+    func._fullname = info.fullname() + '.' + method_name
+    info.names[method_name] = SymbolTableNode(MDEF, func)
+
+
+def attr_s_callback(ctx: ClassDefContext) -> None:
+    """Add an __init__ method to classes decorated with attr.s."""
+    info = ctx.cls.info
+    # TODO: Handle default arguments
+    has_default = {}  # type: Dict[str, Expression]
+    args = []
+
+    for name, table in info.names.items():
+        if table.type:
+            var = Var(name.lstrip("_"), table.type)
```

Comment:
Variable name ``table`` is a bit misleading, I would call this ``symbol`` or similar.